lib/streamlit/elements/alert.py

```diff
@@ -12,86 +12,173 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import cast, TYPE_CHECKING
+from typing import cast, Optional, TYPE_CHECKING
 
 from streamlit.proto.Alert_pb2 import Alert as AlertProto
+from streamlit.errors import StreamlitAPIException
 from .utils import clean_text
+import re
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
     from streamlit.type_util import SupportsStr
 
+MATCH_EMOJI = re.compile(
+    "["
+    u"\U0001F600-\U0001F64F"  # emoticons
+    u"\U0001F300-\U0001F5FF"  # symbols & pictographs
+    u"\U0001F680-\U0001F6FF"  # transport & map symbols
+    u"\U0001F1E0-\U0001F1FF"  # flags (iOS)
+    u"\U00002500-\U00002BEF"  # chinese char
+    u"\U00002702-\U000027B0"
+    u"\U00002702-\U000027B0"
+    u"\U000024C2-\U0001F251"
+    u"\U0001f926-\U0001f937"
+    u"\U00010000-\U0010ffff"
+    u"\u2640-\u2642"
+    u"\u2600-\u2B55"
+    u"\u200d"
+    u"\u23cf"
+    u"\u23e9"
+    u"\u231a"
+    u"\ufe0f"  # dingbats
+    u"\u3030"
+    "]+",
+    flags=re.UNICODE,
+)
+
+# Function to check the icon parameter on the alert.
+# We check if what's been added is a valid emoji,
+# and default to an empty string if not.
+def check_emoji(emoji):
+    extracted_emoji = MATCH_EMOJI.match(emoji)
+
+    # If there's no emoji, carry on
+    if emoji == "":
+        return clean_text(str(""))
+    
+    # If the regex threw a valid result
+    elif extracted_emoji is not None:
+        return clean_text(str(extracted_emoji.group()))
+    
+    # If the regex threw an invalid result
+    else:
+        raise StreamlitAPIException(f'The value "{emoji}" is not a valid emoji. Shortcodes are not allowed, please use a single character instead.')
 
 class AlertMixin:
-    def error(self, body: "SupportsStr") -> "DeltaGenerator":
+    def error(
+        self,
+        body: "SupportsStr",
+        *, # keyword-only args: 
+        icon: Optional[str] = "",
```

Comment:
Not sure if we want "" or None to be default. None feels cleaner?

Reply:
Makes sense! Updated on [3af5a50](https://github.com/streamlit/streamlit/pull/4964/commits/3af5a503ee03c158dee60c8fec28f5c2d55bc348)

Reply:
Sorry, I meant this as something you should discuss with the PM for the project (Johannes?). I don't actually know what's best here 😆